WordPress/gutenberg
# Update the Spinner design

This PR updates the Spinner component based on the designs in #36725. It also adjusts/fixes the position of the Spinner that appears when searching for documents in the add-link interface. Finally, it adds the gray-300 value to color-values.js, which has until now been missing. On this point I did notice that gray-200 has the incorrect value, but as I'm not sure where that is used, I figured we should fix it separately.

Things seem to be working well in the following scenarios:

* Loading a document that has internal dependencies like template parts, or the site logo
* Embedding third party content like a Tweet.
* Searching blocks in the Inserter
* Searching in the link input

<img src="https://user-images.githubusercontent.com/846565/149160764-8e0cbc4c-79eb-423c-bc7c-ec5f622fe993.gif" width="600" />


### Testing instructions

* Perform any action that invokes a spinner and check that it displays and behaves as expected, examples:
  * Edit a template in the site editor that includes template parts
  * Embed third party content like a tweet
  * Search for a block in the Inserter
  * Add a link and search for a post/page
  * Others that I'm missing? :)
* Try different browsers. This is working in Chrome, Firefox, and Safari, but I am unable to test Edge
